packages/web/src/components/gcds-date-input/gcds-date-input.tsx

```diff
@@ -0,0 +1,544 @@
+        this.monthValue = splitValue[1];
+        this.dayValue = splitValue[2];
+      }
+    }
+  }
+
+  /**
+   * Format day input value to add 0 to single digit values
+   */
+  private formatDay(e) {
+    if (!isNaN(e.target.value) && e.target.value.length === 1) {
+      this.dayValue = '0' + e.target.value;
+    }
+  }
+
+  private validateRequiredProps() {
+    this.validateName();
+    this.validateLegend();
+    this.validateFormat();
+
+    if (
+      this.errors.includes('name') ||
+      this.errors.includes('legend') ||
+      this.errors.includes('format')
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  async componentWillLoad() {
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+
+    this.validateValidator();
+
+    // Assign required validator if needed
+    requiredValidator(this.el, 'date-input');
+
+    if (this.validator) {
+      this._validator = getValidator(this.validator);
+    }
+
+    let valid = this.validateRequiredProps();
+
+    if (!valid) {
+      logError('gcds-date-input', this.errors);
+    }
+
+    this.validateValue();
+    if (this.value && isValidDate(this.value)) {
+      this.splitFormValue();
+      this.setValue();
+
+      this.initialValue = this.value;
+    }
+  }
+
+  componentWillUpdate() {
+    if (this.validator) {
+      this._validator = getValidator(this.validator);
+    }
+  }
+
+  render() {
+    const {
+      legend,
+      name,
+      format,
+      required,
+      hint,
+      errorMessage,
+      disabled,
+      lang,
+      hasError,
+    } = this;
+
+    let requiredAttr = {};
+
+    if (required) {
+      requiredAttr['aria-required'] = 'true';
+    }
+
+    const month = (
+      <gcds-select
+        label={i18n[lang].month}
+        selectId="month"
+        name="month"
+        defaultValue={i18n[lang].selectmonth}
+        disabled={disabled}
+        onInput={e => this.handleInput(e, 'month')}
+        onChange={e => this.handleInput(e, 'month')}
+        value={this.monthValue}
+        class={`gcds-date-input__month ${hasError['month'] ? 'gcds-date-input--error' : ''}`}
+        {...requiredAttr}
+        aria-invalid={hasError['month'].toString()}
+        aria-description={hasError['month'] && errorMessage}
+      >
+        <option value="01">{i18n[lang].january}</option>
+        <option value="02">{i18n[lang].february}</option>
+        <option value="03">{i18n[lang].march}</option>
+        <option value="04">{i18n[lang].april}</option>
+        <option value="05">{i18n[lang].may}</option>
+        <option value="06">{i18n[lang].june}</option>
+        <option value="07">{i18n[lang].july}</option>
+        <option value="08">{i18n[lang].august}</option>
+        <option value="09">{i18n[lang].september}</option>
+        <option value="10">{i18n[lang].october}</option>
+        <option value="11">{i18n[lang].november}</option>
+        <option value="12">{i18n[lang].december}</option>
```

Comment:
Could we remove some hard coded values?

```javascript
const options = Array.from({ length: 13 }, (_, i) => (i < 10 ? `0${i}` : `${i}`));

const item = (
  <gcds-select>
    {options.map((option) => (
      <option key={option} value={option}>
        {i18n[lang][months][option]}
      </option>
    ))}
  </select>
);
```

And the `i18n.js` file:
```json
{
  "months": {
    "01": "January",
    "02": "February",
    "03": "March",
    "04": "April",
    "05": "May",
    "06": "June",
    "07": "July",
    "08": "August",
    "09": "September",
    "10": "October",
    "11": "November",
    "12": "December"
  }
}
```

and change the french file accordingly